Validate dependencies

jlpadilla opened the pull request:
Adds validation for the following pre-requisites:
- `bash` version is 4 or greater.
- `jq` is installed.
- `sed` is installed.

jlpadilla commented:
@KevinFCormier

KevinFCormier commented on lib/validateDeps.sh:
```diff
@@ -0,0 +1,32 @@
+# Validate bash version is 4 or greater
+if [ "${BASH_VERSINFO:-0}" -lt 4 ]
+then
+  echo "DEPENDENCY NOT MET. bash version 4 or greater is required to run this tool. Found version ${BASH_VERSION}"
+  echo "On MacOS install with \"brew install bash\".  This bash must be first in your path, but need not be '/bin/bash' or your default login shell."
+  exit 1
```
It might be worth trying to include this code directly in `common.sh` (fairly early on before we run into something that requires bash 4) so you can access the common logging functions. That would make it easier to:
- Make sure all error messages go to stderr
- Prefix errors with `error: ` as is done everywhere else in `ck` (see `error`, `fatal`, `verbose` functions... I would probably use `error` for the main error message, and then `verbose 0` for the text explaining how to recover)

KevinFCormier reviewed (changes requested):
Thanks @jlpadilla! It's great that you don't actually invoke any of the dependencies to check them, so it shouldn't slow things down any noticeable amount.

I'm just requesting a few changes to handle the error messages more similarly to how I've done it in the rest of the code.

jlpadilla commented:
@KevinFCormier I felt `common.sh` was getting too big, so I refactored the logging functions to `lib/logging.sh`.  Updated the validation messages to use the common logging.
